qa/tasks/cephadm.py

```diff
@@ -1964,19 +1964,20 @@ def task(ctx, config):
                             "section.")
         sha1 = config.get('sha1')
         flavor = config.get('flavor', 'default')
+        distro_suffix = config.get('distro-suffix', None)
 
         if any(_ in container_image_name for _ in (':', '@')):
             log.info('Provided image contains tag or digest, using it as is')
             ctx.ceph[cluster_name].image = container_image_name
-        elif sha1:
-            if flavor == "crimson-debug" or flavor == "crimson-release":
-                ctx.ceph[cluster_name].image = container_image_name + ':' + sha1 + '-' + flavor
-            else:
-                ctx.ceph[cluster_name].image = container_image_name + ':' + sha1
-            ref = sha1
         else:
-            # fall back to using the branch value
+            if sha1:
+                ref = sha1
             ctx.ceph[cluster_name].image = container_image_name + ':' + ref
+
+            if distro_suffix is not None:
+                ctx.ceph[cluster_name].image += '-' + distro_suffix
+            if flavor == "crimson-debug" or flavor == "crimson-release":
+                ctx.ceph[cluster_name].image += '-' + flavor
```

Comment:
Generally looks OK to me. I am just curious if anyone else has any opinions on being able to (easily) split the tag into parts, in that case joining with a `-` conflicts a little with `crimson-release`. Should we consider joining the tag with dots `.` or `+` instead?